pkgs/by-name/he/hello-cpp/package.nix

```diff
@@ -0,0 +1,11 @@
+{
+  cmake,
+  ninja,
+  stdenv,
+}:
+
+stdenv.mkDerivation {
+  name = "hello-cpp";
+  src = ./.;
```

Comment:
this will include it's own .nix file which is a pattern that is getting eliminated in nixpkgs.
https://github.com/NixOS/nixpkgs/issues/301014

Reply:
more specifically, you'll probably want to use https://nix.dev/tutorials/working-with-local-files#union-explicitly-include-files

Reply:
Done (but by moving into its own src directory) https://github.com/NixOS/nixpkgs/pull/324905#issuecomment-2212398589